Common/Securities/Option/StrategyMatcher/UnmatchedPositionCountOptionStrategyMatchObjectiveFunction.cs

```diff
@@ -0,0 +1,43 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Linq;
+
+namespace QuantConnect.Securities.Option.StrategyMatcher
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IOptionStrategyMatchObjectiveFunction"/> that evaluates the number of unmatched
+    /// positions, in number of contracts, giving precedence to solutions that have fewer unmatched contracts.
+    /// </summary>
+    public class UnmatchedPositionCountOptionStrategyMatchObjectiveFunction : IOptionStrategyMatchObjectiveFunction
```

Comment:
Right now we only have a `MatchOnce` in the matcher. In order to 'optimize' we would generate multiple match solutions and evaluate each one with the objective function, picking the best one. The natural fit for an objective function would be the margin required for the position. Since this PR is complex enough, I left optimizing the strategy match solutions for later. It may not be needed and is only likely to be useful if there are many option positions for a single underlying. We _could_ just as well remove this, but I felt it valuable to leave in as documentation of intent. I can add a blurb to the matcher about its intended purposes/function.

Reply:
There actually is already a comment in the matcher:

``` csharp
// TODO : Implement matching multiple permutations and using the objective function to select the best solution
```

This would be another match function which would evaluate `MatchOnce` multiple times, each time forcing a different permutation of matches yield _n_ different solutions which would each be evaluated w/ the objective function and then we would pick the best.